improver/nbhood/recursive_filter.py

```diff
@@ -304,50 +304,56 @@ def _validate_smoothing_coefficients(cube, smoothing_coefficients_cube):
             ValueError: The coordinate to be smoothed within the
                 smoothing coefficient cube does not have the expected points.
         """
-        if smoothing_coefficients_cube.name() not in [
-            "smoothing_coefficient_x",
-            "smoothing_coefficient_y",
-        ]:
+        if smoothing_coefficients_cube.name() == "smoothing_coefficient_x":
+            smoothing_axis = "x"
+            non_smoothing_axis = "y"
+            smoothing_coord = smoothing_coefficients_cube.coord(axis="x")
+            non_smoothing_coord = smoothing_coefficients_cube.coord(axis="y")
```

Comment:
If you change these two lines like this, then you can refactor them to be after the else block and remove the repetition in the elif block.
```suggestion
            smoothing_coord = smoothing_coefficients_cube.coord(axis=smoothing_axis)
            non_smoothing_coord = smoothing_coefficients_cube.coord(axis=non_smoothing_axis)
```